sdk/appcenter-distribute/src/main/java/com/microsoft/appcenter/distribute/DistributeUtils.java

```diff
@@ -129,23 +141,31 @@ static void updateSetupUsingBrowser(Activity activity, String installUrl, String
         /* Generate request identifier. */
         String requestId = UUID.randomUUID().toString();
 
-        /* Build URL. */
-        String url = installUrl;
-        url += String.format(UPDATE_SETUP_PATH_FORMAT, appSecret);
-        url += "?" + PARAMETER_RELEASE_HASH + "=" + releaseHash;
-        url += "&" + PARAMETER_REDIRECT_ID + "=" + activity.getPackageName();
-        url += "&" + PARAMETER_REDIRECT_SCHEME + "=" + "appcenter";
-        url += "&" + PARAMETER_REQUEST_ID + "=" + requestId;
-        url += "&" + PARAMETER_PLATFORM + "=" + PARAMETER_PLATFORM_VALUE;
-        url += "&" + PARAMETER_ENABLE_UPDATE_SETUP_FAILURE_REDIRECT_KEY + "=" + "true";
-        url += "&" + PARAMETER_INSTALL_ID + "=" + AppCenter.getInstallId().get().toString();
-        AppCenterLog.debug(LOG_TAG, "No token, need to open browser to url=" + url);
-
         /* Store request id. */
         SharedPreferencesManager.putString(PREFERENCE_KEY_REQUEST_ID, requestId);
 
-        /* Open browser, remember that whatever the outcome to avoid opening it twice. */
-        BrowserUtils.openBrowser(url, activity);
+        /* Build URL. */
+        final StringBuilder urlBuilder = new StringBuilder(installUrl);
+        urlBuilder.append(String.format(UPDATE_SETUP_PATH_FORMAT, appSecret));
+        urlBuilder.append("?" + PARAMETER_RELEASE_HASH + "=").append(releaseHash);
+        urlBuilder.append("&" + PARAMETER_REDIRECT_ID + "=").append(activity.getPackageName());
+        urlBuilder.append("&" + PARAMETER_REDIRECT_SCHEME + "=" + "appcenter");
+        urlBuilder.append("&" + PARAMETER_REQUEST_ID + "=").append(requestId);
+        urlBuilder.append("&" + PARAMETER_PLATFORM + "=" + PARAMETER_PLATFORM_VALUE);
+        urlBuilder.append("&" + PARAMETER_ENABLE_UPDATE_SETUP_FAILURE_REDIRECT_KEY + "=" + "true");
+        AppCenter.getInstallId().thenAccept(new AppCenterConsumer<UUID>() {
+
+            @Override
+            public void accept(UUID uuid) {
+                urlBuilder.append("&" + PARAMETER_INSTALL_ID + "=").append(uuid.toString());
+                String url = urlBuilder.toString();
+                AppCenterLog.debug(LOG_TAG, "No token, need to open browser to url=" + url);
+
+
+                /* Open browser, remember that whatever the outcome to avoid opening it twice. */
+                BrowserUtils.openBrowser(url, activity);
```

Comment:
Use the application context and flag new task, since the API is async the activity might be finishing or finished after callback.